concepts/classes/introduction.md

```diff
@@ -0,0 +1,95 @@
+# Introduction
+
+It is time to get to one of the core principles of C++: object-oriented programming (OOP).
+OOP is centered around `classes` - user-defined types of data with their own set of related functions.
+We will start with the basics and will cover more advanced topics further down the syllabus tree.
+
+## Members
+
+Classes can have **member variables** and **member functions**.
+They are accessed by the **member selection** operator `.`.
+Just as variables outside of `classes`, it is advisable to initialize member variables with a value upon declaration.
+This value will then become the default for newly created objects from this class.
+
+## Encapsulation and Information Hiding
+
+Classes offer the option to restrict access to their members.
+The two basic cases are `private` and `public`.
+All members are `private` by default and need to be explicitly marked to be usable outside of the class.
+
+## Basic example
+
+The definition of a new `class` can be seen in the following example.
+Notice the `;` after the definition.
+
+```cpp
+class Wizzard {
+  public:
+    int cast_spell() {
+      return damage;
+    }
+    std::string name{};
+  private:
+    int damage{5};
+};
+
```

Comment:
You could consider adding comments to this code example, explaining key lines.
E.g. something like
```
  public:   // from here on all members are accessible publicly
    inc cast_spell() {  // defines the public member function cast_spell
    ...
    std::string name{};  // defines the public member variable `name`
  private:  // from here on all members are private
    int damage{5};  // defines the private member variable `damage`
```

Reply:
Good idea! This would help indeed